src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -356,93 +447,427 @@ bool IsCreateHelper(IMethodSymbol method)
+            INamedTypeSymbol localizableStringType,
+            INamedTypeSymbol localizableResourceStringType)
         {
             IArgumentOperation titleArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("title", StringComparison.OrdinalIgnoreCase));
-            if (titleArgument?.Parameter.Type.SpecialType != SpecialType.System_String)
+            if (titleArgument != null)
             {
-                return;
+                if (titleArgument.Parameter.Type.SpecialType == SpecialType.System_String)
+                {
+                    operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+                }
+
+                AnalyzeDescriptorArgument(operationAnalysisContext, titleArgument,
+                    AnalyzeTitleCore, containingType, localizableTitles, resourceDataValueMap,
+                    localizableStringType, localizableResourceStringType);
             }
+        }
+
+        private static void AnalyzeTitleCore(string title, IArgumentOperation argumentOperation, Location fixLocation, Action<Diagnostic> reportDiagnostic)
+        {
+            var isMultiSentences = IsMultiSentences(title);
+            var endsWithPeriod = EndsWithPeriod(title);
+            var containsLineReturn = ContainsLineReturn(title);
+            if (isMultiSentences || endsWithPeriod || containsLineReturn)
+            {
+                var fixedTitle = endsWithPeriod ? RemoveTrailingPeriod(title) : title;
+                fixedTitle = isMultiSentences ? FixMultiSentences(fixedTitle) : fixedTitle;
+                fixedTitle = containsLineReturn ? FixLineReturns(fixedTitle, allowMultisentences: false) : fixedTitle;
+                Debug.Assert(title != fixedTitle);
 
-            operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+                ReportDefineDiagnosticArgumentCorrectlyDiagnostic(DefineDiagnosticTitleCorrectlyRule,
+                    argumentOperation, fixedTitle, fixLocation, reportDiagnostic);
+            }
+        }
 
-            if (!TryGetNonEmptyConstantStringValue(titleArgument, out var title))
+        private static void ReportDefineDiagnosticArgumentCorrectlyDiagnostic(
+            DiagnosticDescriptor descriptor,
+            IArgumentOperation argumentOperation,
+            string fixValue,
+            Location fixLocation,
+            Action<Diagnostic> reportDiagnostic)
+        {
+            // Additional location in an additional document does not seem to be preserved
+            // from analyzer to code fix due to a Roslyn bug.
+            // We workaround this bug by passing additional document file path and location span as strings.
+            // TODO: File a Roslyn bug.
```

Comment:
Open TODO without link.

Reply:
Yep, I need to file a Roslyn issue.

Reply:
https://github.com/dotnet/roslyn/issues/46377